enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/service/CommonOtpService.java

```diff
@@ -172,4 +184,30 @@ private void handleFailedOtpVerification(OnboardingProcessEntity process, OwnerI
         onboardingProcessRepository.save(process);
     }
 
+    /**
+     * Handle maximum failed attempts for OTP verification based on OTP type.
+     * @param process Onboarding process.
+     * @param otp Onboarding OTP.
+     * @param ownerId Owner identification.
+     * @return Updated onboarding process entity.
+     */
+    private OnboardingProcessEntity handleOtpMaxFailedAttempts(OnboardingProcessEntity process, OnboardingOtpEntity otp, OwnerId ownerId) {
+        if (otp.getType() == OtpType.USER_VERIFICATION) {
+            // Reset current identity verification, if possible. The process may
+            try {
+                verificationLimitService.resetIdentityVerification(ownerId);
+            } catch (RemoteCommunicationException | IdentityVerificationException | OnboardingProcessLimitException | OnboardingProcessException ex) {
+                logger.error("Identity verification reset failed, error: " + ex.getMessage(), ex);
+                // Obtain most current process entity, the process may have failed due to reached limit of identity verification resets
+                Optional <OnboardingProcessEntity> updatedProcessOptional = onboardingProcessRepository.findById(process.getId());
+                if (updatedProcessOptional.isPresent()) {
```

Comment:
🤔 So may it happen that the process is not found and we return the old reference? Shouldn't we throw an exception?

Reply:
This is basically impossible, but ok, we can handle it.